src/components/InLineNotification.tsx

```diff
@@ -0,0 +1,158 @@
+import React from 'react'
+import { StyleProp, StyleSheet, Text, TextStyle, View, ViewStyle } from 'react-native'
+import AttentionIcon from 'src/icons/Attention'
+import Colors from 'src/styles/colors'
+import fontStyles, { typeScale } from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+
+export enum Severity {
+  Informational,
+  Warning,
+  Error,
+}
+
+interface Props {
+  severity: Severity
+  title?: string | null
+  description: string
+  style?: StyleProp<ViewStyle>
+  ctaLabel?: string | null
+  onPressCta?: () => void
+  ctaLabel2?: string | null
+  onPressCta2?: () => void
+  testID?: string
+}
+
+interface CustomStyles {
+  container: ViewStyle
+  attentionIcon: Record<'color', Colors>
+  ctaLabel: TextStyle
+  ctaLabel2: TextStyle
+}
+
+export function InLineNotification({
+  severity,
+  title,
+  description,
+  style,
+  ctaLabel,
+  onPressCta,
+  ctaLabel2,
+  onPressCta2,
+  testID,
+}: Props) {
+  const severityStyle = severityStyles[severity]
+  return (
+    <View style={[defaultStyles.container, severityStyle.container, style]} testID={testID}>
+      {title && (
+        <View style={defaultStyles.row}>
+          <View style={defaultStyles.attentionIcon}>
+            <AttentionIcon color={severityStyle.attentionIcon.color} />
+          </View>
+          <Text style={defaultStyles.titleText}>{title}</Text>
+        </View>
+      )}
+
+      <View style={defaultStyles.bodyContainer}>
+        <View style={[defaultStyles.row, defaultStyles.bodyRow]}>
+          <View style={defaultStyles.attentionIcon}>
+            {!title && <AttentionIcon color={severityStyle.attentionIcon.color} />}
+          </View>
+          <Text style={defaultStyles.bodyText}>{description}</Text>
+        </View>
+
+        {(ctaLabel || ctaLabel2) && (
+          <View style={[defaultStyles.row, defaultStyles.ctaRow]}>
+            {ctaLabel && onPressCta && (
+              <Text style={[defaultStyles.ctaLabel, severityStyle.ctaLabel]} onPress={onPressCta}>
+                {ctaLabel}
+              </Text>
+            )}
+            {ctaLabel2 && onPressCta2 && (
+              <Text
+                style={[defaultStyles.ctaLabel2, severityStyle.ctaLabel2]}
+                onPress={onPressCta2}
+              >
+                {ctaLabel2}
+              </Text>
+            )}
+          </View>
+        )}
+      </View>
+    </View>
+  )
+}
+
+const defaultStyles = StyleSheet.create({
+  container: {
+    paddingHorizontal: Spacing.Regular16,
+    paddingTop: Spacing.Regular16,
+    paddingBottom: Spacing.Small12,
+    borderRadius: Spacing.Regular16,
+    width: '100%',
+    gap: Spacing.Tiny4,
+  },
+  bodyContainer: {
+    gap: Spacing.Smallest8,
+  },
+  row: {
+    flexDirection: 'row',
+    gap: Spacing.Smallest8,
+    alignItems: 'center',
+  },
+  bodyRow: {
+    alignItems: 'flex-start',
```

Comment:
this is already the default value, do you need this?

Reply:
It's not necessary for the text component but for the row's View component. It groups both the icon's View and Text components together.

If the description is spread across 2 lines or more, then the icon would be vertically centered. This would not look great. Here's a preview:

![image](https://github.com/valora-inc/wallet/assets/147835134/cc38fbd3-6370-493a-a6e9-5c7d226283cf)

Reply:
is it a hard requirement that the icon is vertically centered if the title exceeds one line? there's no example in the Figma file, and the example with the multi-line description + no title shows the icon is not vertically centered. so if it is okay to keep this pattern so that the icon is always aligned to the top, i have an idea to simplify the styles of this component. i tried to describe this [here](https://github.com/valora-inc/wallet/pull/4436#discussion_r1387808143) but i think i was not clear, i made a [branch](https://github.com/valora-inc/wallet/compare/denvalora/notifications...kathy/notification-alternative?expand=1) to show you my idea - lmk what you think :)

Reply:
I've cherry picked your commit and added it to this branch.

## Before
| Description | Description + CTA | Short title + description | Title + description | Title + description + CTA |
| -- | -- | -- | -- | -- |
| ![image](https://github.com/valora-inc/wallet/assets/147835134/66741394-0166-4097-81e4-5dd1fc1b220a) | ![image](https://github.com/valora-inc/wallet/assets/147835134/b069b1e0-ff13-444a-814b-0e58b1a25fcc) | ![image](https://github.com/valora-inc/wallet/assets/147835134/e21c64b3-ab96-4ddb-a36d-acfdfa6e0301) | ![image](https://github.com/valora-inc/wallet/assets/147835134/0615e9ba-ff8a-4ff1-aa2e-68979b0add9e) | ![image](https://github.com/valora-inc/wallet/assets/147835134/29e715ff-b06f-448f-9225-412b3b2e7576) |


## After
| Description | Description + CTA | Short title + description | Title + description | Title + description + CTA |
| -- | -- | -- | -- | -- |
| ![image](https://github.com/valora-inc/wallet/assets/147835134/6c9198a9-7956-43de-bf5e-7d30a5b2fc43) | ![image](https://github.com/valora-inc/wallet/assets/147835134/9647f05b-4649-4120-9364-304e7446c43e) | ![image](https://github.com/valora-inc/wallet/assets/147835134/6643a620-7c03-42ae-af3c-7a9a74978976) | ![image](https://github.com/valora-inc/wallet/assets/147835134/75d90652-6bee-4a9c-8421-09d987b0cbe0) | ![image](https://github.com/valora-inc/wallet/assets/147835134/d6b54306-799d-4501-8bb4-44375e26e4ec) |